include/s3select_oper.h

```diff
@@ -2256,7 +2263,17 @@ class derive_x : public base_time_to_string
   public:
     std::string print_time(boost::posix_time::ptime& new_ptime, boost::posix_time::time_duration& td, uint32_t param)
     {
-      return std::to_string(td.hours());
+      int tz_minute = td.minutes();
+      std::string tz_hr = std::to_string(std::abs(td.hours()));
+      if (tz_minute == 0)
+      {
+        return (td.is_negative() ? "-" : "+") + std::string(2 - tz_hr.length(), '0') + tz_hr;
+      }
+      else
+      {
+        std::string tz_mn = std::to_string(std::abs(tz_minute));
+        return (td.is_negative() ? "-" : "+") + std::string(2 - tz_hr.length(), '0') + tz_hr + std::string(2 - tz_mn.length(), '0') + tz_mn;
```

Comment:
the print_time is constructing and destructing many temporary std::string (its called intensively).

no need to change that now, but we should later test these parts for performance improvements.